### PR TITLE
87 possibility to add more copy of the same card

### PR DESCRIPTION
**Pull request by Bjolidon**
You can now add more copy of the same card in your collection through the single card fragment. You may have to use deleteDatabase from LocalDatabaseProvider once to make it work as I modified DBMagicCard

### CODE REVIEW

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserRTDB.kt [hunk 1]
```diff
@@ -38,12 +67,68 @@ class UserRTDB(database: Database) { //Firebase.database.reference.child("users"
...
+     * Removes a list of cards from the users collection.
+     */
+    fun removeAllCopyOfCard(userUID: String, fbcard: DBMagicCard) {
+        val cardUID = fbcard.getFbKey()
+        val fbpossession = fbcard.possession.toString().lowercase()
+        val cardRef = db.child(userUID).child(fbpossession).child(cardUID)
+        cardRef.runTransaction(object : Transaction.Handler {
+            override fun doTransaction(currentData: MutableData): Transaction.Result {
+                val currentValue = currentData.getValue(Int::class.java)
+                if (currentValue != null) {
```
Comment (Cocograal) [inline app/src/main/java/com/github/sdp/tarjetakuna/database/UserRTDB.kt:112]: What is the point of doing this? either way you set the value to 0?

**Review by Cocograal — CHANGES_REQUESTED** [review]

**Review by Cocograal — DISMISSED** [review]
Thanks for the changes

File: app/src/main/res/layout/cards_recycler_view_row.xml [hunk 2]
```diff
@@ -52,6 +52,19 @@
                 android:text="@string/m15"
                 app:layout_constraintStart_toEndOf="@+id/cardImageView"
                 app:layout_constraintTop_toBottomOf="@+id/cardNameRecyclerViewTextView" />
+
+            <TextView
+                android:id="@+id/quantityRecyclerViewTextView"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginEnd="10dp"
```
Comment (KelvinKappeler) [inline app/src/main/res/layout/cards_recycler_view_row.xml:60]: Do you need maybe to use things in style.xml and not hardcoded 10dp?

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/SingleCardManageCollectionTest.kt [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        // check that the quantity displayed is the one in the database
+        cardQuantityText.check(matches(withText("2")))
+    }
+
+    @Test
+    fun cardQuantityWhenAddingCardIsCorrect() {
+        addCardButton.perform(click())
+        cardQuantityText.check(matches(withText("3")))
+        addCardButton.perform(click())
+        cardQuantityText.check(matches(withText("4")))
```
Comment (KelvinKappeler) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/SingleCardManageCollectionTest.kt:112]: Maybe don't use hard-coded values here?

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/BrowserFragmentTest.kt [hunk 4]
```diff
@@ -265,6 +271,17 @@ class BrowserFragmentTest {
         ).check(matches(hasDescendant(withText("Pégase solgrâce"))))
     }
 
+    @Test
+    fun multipleCopyOfTheSameCard() {
+        onView(withId(R.id.browser_searchbar)).perform(click())
+        onView(withId(R.id.browser_searchbar)).perform(typeText("Ambush Paratrooper 05"))
```
Comment (KelvinKappeler) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/BrowserFragmentTest.kt:277]: Maybe don't use hard-coded values here?

**Review by KelvinKappeler — CHANGES_REQUESTED** [review]
I don't have much to say. Congratulations on your work, well done! However, please correct the tests in order to be able to merge in the main without problems (and to check the total coverage).

In addition, it does not seem to me that you have tested the new MagicCard with the quantity added

**Comment by Bjolidon** [comment]
I added checks in my SingleCardFragment tests to check if the quantity of the DBMagicCards are updated accordingly

**Review by KelvinKappeler — APPROVED** [review]
Thanks for your changes!

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/SingleCardManageCollectionTest.kt [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+        runBlocking {
+            withTimeout(5000) {
+                LocalDatabaseProvider.getDatabase(LocalDatabaseProvider.CARDS_DATABASE_NAME)
+                    ?.magicCardDao()?.getCard(card.set.code, card.number.toString())?.let {
+                        assert(it.quantity == 4)
```
Comment (AlSchlo) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/SingleCardManageCollectionTest.kt:120]: Wrong assertion

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/SingleCardManageCollectionTest.kt [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            Bundle().apply {
+                putString(
+                    "card",
+                    Gson().toJson(card)
+                )
+            }
+        scenario = launchFragmentInContainer(fragmentArgs = bundleArgs)
+        scenario.moveToState(Lifecycle.State.STARTED)
+
+        Thread.sleep(1000)
```
Comment (AlSchlo) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/SingleCardManageCollectionTest.kt:84]: 😨

**Pull request merged**